### PR TITLE
feat: Text to bigquery template

### PR DESCRIPTION
**Pull request by hhasija**

### CODE REVIEW

File: java/README.md [hunk 1]
```diff
@@ -9,10 +9,17 @@ Please refer to the [Dataproc Templates (Java - Spark) README](java/README.md)
...
+=======
 * [GCSToBigTable](/java/src/main/java/com/google/cloud/dataproc/templates/gcs#1-gcs-to-bigquery)(blogpost [link](https://medium.com/google-cloud/load-data-from-gcs-to-bigtable-using-gcp-dataproc-serverless-5c43d773e615)) (Video [link](https://www.youtube.com/watch?v=JwO72i2nVPU))
 * [GCSToGCS](/java/src/main/java/com/google/cloud/dataproc/templates/gcs#4-gcs-to-gcs)(blogpost [link](https://medium.com/@ankuljain/migrate-gcs-to-gcs-using-dataproc-serverless-3b7b0f6ad6b9))
 * [GCSToJDBC](/java/src/main/java/com/google/cloud/dataproc/templates/gcs#3-gcs-to-jdbc)(blogpost [link](https://medium.com/google-cloud/importing-data-from-gcs-to-databases-via-jdbc-using-dataproc-serverless-7ed75eab93ba))
 * [GCSToSpanner](/java/src/main/java/com/google/cloud/dataproc/templates/gcs#2-gcs-to-spanner)(blogpost [link](https://medium.com/google-cloud/fast-export-large-database-tables-using-gcp-serverless-dataproc-spark-bb32b1260268))
+>>>>>>> 87c9bf8411f83f8beec6ef5649d321190a6542f1
```
Comment (vanshaj-bhatia) [inline java/README.md:22]: Please resolve the conflicts and remove the placeholders.

File: java/src/main/java/com/google/cloud/dataproc/templates/gcs/TextToBigquery.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            .csv(inputLocation);
+
+    if (bqTempTable != null && bqTempQuery != null) {
+      inputData.createOrReplaceGlobalTempView(bqTempTable);
+      inputData = spark.sql(bqTempQuery);
+    }
+
+    inputData
+        .write()
+        .format("com.google.cloud.spark.bigquery")
```
Comment (vanshaj-bhatia) [inline java/src/main/java/com/google/cloud/dataproc/templates/gcs/TextToBigquery.java:86]: Lets replace com.google.cloud.spark.bigquery with TemplateConstants.SPARK_READ_FORMAT_BIGQUERY

File: java/src/main/java/com/google/cloud/dataproc/templates/gcs/TextToBigquery.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        .option(GCS_BQ_CSV_HEADER, true)
```
Comment (vanshaj-bhatia) [inline java/src/main/java/com/google/cloud/dataproc/templates/gcs/TextToBigquery.java:87]: I think, we do not require GCS_BQ_CSV_HEADER while writing to BQ

File: java/src/main/java/com/google/cloud/dataproc/templates/gcs/TextToBigquery.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+          TEXT_BIGQUERY_OUTPUT_MODE,
+          TEXT_BIGQUERY_TEMP_BUCKET);
+      throw new IllegalArgumentException(
+          "Required parameters for TextToBigquery not passed. "
+              + "Set mandatory parameter for TextToBigquery template "
+              + "in resources/conf/template.properties file.");
+    }
+
+    LOGGER.info(
+        "Starting GCS to GCS spark job with following parameters:"
```
Comment (vanshaj-bhatia) [inline java/src/main/java/com/google/cloud/dataproc/templates/gcs/TextToBigquery.java:122]: Replace GCS to GCS with Text to BQ

**Review by vanshaj-bhatia — CHANGES_REQUESTED** [review]

File: java/src/main/java/com/google/cloud/dataproc/templates/gcs/TextToBigquery.java (hunk 2 above)
Comment (hhasija) [reply java/src/main/java/com/google/cloud/dataproc/templates/gcs/TextToBigquery.java:86]: Done

File: java/README.md (hunk 1 above)
Comment (hhasija) [reply java/README.md:22]: Done

**Review by vanshaj-bhatia — APPROVED** [review]
LGTM!

**Pull request merged**